code/obj/item.dm

```diff
@@ -378,73 +378,198 @@
+				user, "<span class='alert'>You remove \the [master] from [M]'s face!</span>",\
+				M, "<span class='[is_awful_monsterthing ? "alert" : "notice"]'><b>[user]</b> stops forcing \the [master] down your throat!</span>")
+			else
+				user.tri_message("<span class='alert'><b>[user]</b> stops trying to shove \the [master] down [M]'s throat!</span>",\
+				user, "<span class='alert'>You remove \the [master] from [M]'s face!</span>",\
+				M, "<span class='alert'><b>[user]</b> stops trying to shove \the [master] down your throat!</span>")
+
+	onUpdate()
+		..()
+		if(src.failchecks())
+			interrupt(INTERRUPT_ALWAYS)
+
+	onEnd()
+		..()
+		if(src.failchecks())
+			interrupt(INTERRUPT_ALWAYS)
+
+		if(src.M_is_user)
+			if(src.is_it_organs)
+				M.visible_message("<span class='alert'>[M] takes a visceral bite out of \the [master][prob(30) ? " like a [pick(src.grody_adj)] [pick(src.grody_noun)]" : ""]!</span>",\
+				"<span class='[is_awful_monsterthing ? "notice" : "alert"]'>You take a bite of [master][(prob(30) && !is_awful_monsterthing )? ", you [pick(src.grody_adj)] [pick(src.grody_noun)]" : ""]!</span>")
+			else
+				M.visible_message("<span class='notice'>[M] takes a bite of \the [master].</span>",\
+				"<span class='notice'>You take a bite of [master].</span>")
 
-		user.tri_message("<span class='alert'><b>[user]</b> feeds [M] [src]!</span>",\
-		user, "<span class='alert'>You feed [M] [src]!</span>",\
-		M, "<span class='alert'><b>[user]</b> feeds you [src]!</span>")
-		logTheThing("combat", user, M, "feeds [constructTarget(M,"combat")] [src] [log_reagents(src)]")
+		else
+			if(src.is_it_organs)
+				user.tri_message("<span class='alert'><b>[user]</b> forces a chunk of [master] down [M]'s throat[prob(30) ? " like a [pick(src.grody_adj)] [pick(src.grody_noun)]" : ""]!</span>",\
+				user, "<span class='alert'>You force [M] too eat a chunk of [master]!</span>",\
+				M, "<span class='[is_awful_monsterthing ? "notice" : "alert"]'><b>[user]</b> feeds you [master]!</span>")
+			else
+				user.tri_message("<span class='alert'><b>[user]</b> feeds [M] [master]!</span>",\
+				user, "<span class='alert'>You feed [M] [master]!</span>",\
+				M, "<span class='alert'><b>[user]</b> feeds you [master]!</span>")
+			logTheThing("combat", user, M, "feeds [constructTarget(M,"combat")] [master] [log_reagents(master)]")
 
-		if (src.material && src.material.edible)
-			src.material.triggerEat(M, src)
+		if (master.material && master.material.edible)
+			master.material.triggerEat(M, master)
 
-		if (src.reagents && src.reagents.total_volume)
-			src.reagents.reaction(M, INGEST)
-			SPAWN_DBG(0.5 SECONDS) // Necessary.
-				src.reagents.trans_to(M, src.reagents.total_volume)
+		if (master.reagents && master.reagents.total_volume)
+			master.reagents.reaction(M, INGEST)
+			master.reagents.trans_to(M, master.reagents.total_volume/master.amount)
 
-		playsound(M.loc, "sound/items/eatfood.ogg", rand(10, 50), 1)
+		if (!master || !M || !user)
+			return
```

Comment:
why doing this and not failchecks which has this - we have failchecks earlier anyways

Reply:
Whoops, missed that.